habito/habito.py

```diff
@@ -140,6 +140,41 @@ def add(name, quantum, units):
                       created_date=datetime.now(), quantum=quantum,
                       units=units, magica="")
 
+@cli.command()
+@click.argument("id", type=click.INT)
+@click.option('--name', '-n', help="The new name (leave empty to leave unchanged)")
+@click.option('--quantum', '-q', help="The new quantum (leave empty to leave unchanged)", type=click.FLOAT)
+def edit(id, name, quantum):
+    try:
+        habit = HabitModel.get(HabitModel.id == id)
+    except DoesNotExist:
+        click.echo("The habit you're trying to edit does not exist!")
+        raise SystemExit(1)
+    habit.name = name.strip() or habit.name
+    habit.quantum = quantum or habit.quantum
+    habit.save()
+    click.echo("Habit with id {} has been saved with name: {} and quantum: {}".format(id, habit.name, habit.quantum))
+
+
+@cli.command()
+@click.argument("id", type=click.INT)
+@click.option("--keeplogs", is_flag=True, default=False)
+def delete(id, keeplogs):
```

Comment:
Makes sense. The delete with `keeplogs` and archive appear similar in
purpose. Can we merge them? May be `delete --archive` just turns off the
active flag.

Agree with you, privacy is not in immediate scope for the tool.

On Sun, May 22, 2016 at 3:53 PM, Ben Baert <notifications@github.com> wrote:

> In habito/habito.py
> <https://github.com/codito/habito/pull/4#discussion_r64147619>:
>
> > +def edit(id, name, quantum):
> > +    try:
> > +        habit = HabitModel.get(HabitModel.id == id)
> > +    except DoesNotExist:
> > +        click.echo("The habit you're trying to edit does not exist!")
> > +        raise SystemExit(1)
> > +    habit.name = name.strip() or habit.name
> > +    habit.quantum = quantum or habit.quantum
> > +    habit.save()
> > +    click.echo("Habit with id {} has been saved with name: {} and quantum: {}".format(id, habit.name, habit.quantum))
> > +
> > +
> > +@cli.command()
> > +@click.argument("id", type=click.INT)
> > +@click.option("--keeplogs", is_flag=True, default=False)
> > +def delete(id, keeplogs):
>
> I think it's possible to simply delete a habit added by mistake, so I
> think the ability to delete a habit is still useful. It's also possible
> that you add habits to experiment with the program, so it's useful to be
> able to delete those again. I did this when I first installed the software
> and then I was stuck with random habits that I couldn't remove without
> accessing the database directly. :)
>
> Having said this, I definitely like your idea of an active field so that
> past habits are still in the database, but not necessarily shown in the
> list anymore. This can also be helpful in future features, for instance
> generating a report of the past month/year/...: (e.g. in the past year, I
> built the habit of writing 1000 words a day and running 5 kilometres a
> day).
>
> As for the privacy issues you mentioned, I think we might want to give the
> option to encrypt the database and require a password per session. I would
> *definitely* not do this by default, as I think most users don't require
> it, but it's a good option. I don't think it's a very high priority right
> now, but in the long run, it can definitely be a useful feature.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/codito/habito/pull/4/files/76791bae2ca317d3b0c07f7bbf3a9d038b0d64b4#r64147619>
>